File: output/gemma4-e4b/avatar/Python/Java/codeforces_653_A.java
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class codeforces_653_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read N (the first integer, which is unused in the logic but required for input structure)
        if (!scanner.hasNextInt()) {
            System.out.println("NO");
            scanner.close();
            return;
        }
        int n = scanner.nextInt();

        // Use a Set to store the unique elements for O(1) lookups
        Set<Integer> set = new HashSet<>();
        
        // Read all remaining integers from the input stream into the set
        while (scanner.hasNextInt()) {
            set.add(scanner.nextInt());
        }
        
        scanner.close();

        boolean found = false;
        
        // Iterate through the unique elements in the set
        for (int i : set) {
            // Check if i+1 and i+2 are also present in the set
            if (set.contains(i + 1) && set.contains(i + 2)) {
                found = true;
                break;
            }
        }

        if (found) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }
}

File: output/gemma4-e4b/avatar/Java/Python/codeforces_653_A.py
import sys

def solve():
    # Read all input data from stdin
    data = sys.stdin.read().split()
    
    if not data:
        print("NO")
        return

    try:
        # The first element is N
        N = int(data[0])
        # The remaining N elements are the list values
        if N > 0:
            A_list = [int(x) for x in data[1:N+1]]
        else:
            A_list = []
    except IndexError:
        # Handle case where input might be incomplete
        print("NO")
        return

    # Get unique and sorted elements
    A = sorted(list(set(A_list)))

    # Check for three consecutive numbers
    found = False
    if len(A) >= 3:
        for i in range(len(A) - 2):
            # Check if A[i], A[i+1], A[i+2] form a sequence x, x+1, x+2
            if A[i] + 1 == A[i+1] and A[i+1] + 1 == A[i+2]:
                found = True
                break

    print("YES" if found else "NO")

if __name__ == "__main__":
    solve()